app/src/main/java/com/example/androidtrainingproject/ui/home/HomeScreen.kt

```diff
@@ -1,61 +1,106 @@
+import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.RangeSlider
 import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SliderDefaults
 import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawBehind
 import androidx.compose.ui.draw.paint
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.dimensionResource
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
-import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.hilt.navigation.compose.hiltViewModel
-import androidx.navigation.NavController
 import coil.compose.AsyncImage
 import com.example.androidtrainingproject.R
+import com.example.androidtrainingproject.ui.destinations.ProductDetailsDestination
 import com.example.androidtrainingproject.ui.shared.AppTopBar
 import com.example.androidtrainingproject.ui.shared.SearchBox
+import com.example.androidtrainingproject.ui.theme.Black
 import com.example.androidtrainingproject.ui.theme.Gray
 import com.example.androidtrainingproject.ui.theme.LabelText
 import com.example.androidtrainingproject.ui.theme.Purple
+import com.example.androidtrainingproject.ui.theme.White
+import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.annotation.RootNavGraph
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import kotlinx.coroutines.launch
+import kotlin.math.roundToInt
 
+@OptIn(ExperimentalMaterial3Api::class)
+@RootNavGraph(start = true)
+@Destination
 @Composable
-fun Home(navController: NavController) {
+fun Home(destinationsNavigator: DestinationsNavigator) {
     val homeViewModel: HomeViewModel = hiltViewModel()
     val searchText by homeViewModel.searchText.collectAsState()
+    val activeFiltersCount by homeViewModel.activeFiltersCount.collectAsState()
+
+    val coroutineScope = rememberCoroutineScope()
+    val sheetState = rememberModalBottomSheetState()
+
+    var ratingFilter by remember { mutableStateOf(1) }
+    var priceRange by remember { mutableStateOf(0f..200f) }
+    var checkedCategories by remember { mutableStateOf(mutableListOf<String>()) }
+
+    homeViewModel.allCategories.let { categories ->
```

Comment:
This will unnecessarily trigger on every rerender. Better expose it as a getter in the view model itself, i.e:

`val checkedCategories get() = categories.map { category -> category.name } as MutableList<String>`

Edit: Just saw that you assign value to this as well. You could play with a `mutableState` (or ideally `StateFlow`) in the view model and a custom getter + setter for it but I no longer see any need for a getter/setter property. You can just have a regular state and make this calculation in the default `init`.